software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java

```diff
@@ -360,7 +316,7 @@ protected int tryJavaInstall(String version, String command) {
      * Checks for the version of Java installed on the entity's location over SSH.
      * @return An Optional containing the version portion of `java -version`, or absent if no Java found.
      */
-    protected Optional<String> getCurrentJavaVersion() {
+    protected Optional<String> getInstalledJavaVersion() {
```

Comment:
@andreaturli strongly agree, this class is often subclassed so let's ensure we maintain api compatibility even on protected methods.  apart from this good to merge in my view.